### PR TITLE
[Teams] Private IPs & hostnames docs

### PR DESCRIPTION
**Pull request by abracchi-tw**

### CODE REVIEW

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+pcx-content-type: concept
+---
+
+# Private hostnames and IPs
+
+Building out a private network has two primary components: the infrastructure side, and the client side.
+
+The infrastructure side of the equation is powered by Cloudflare Tunnel, which connects your infrastructure to Cloudflare — whether that be a singular application, many applications, or an entire network segment. This is made possible by running `cloudflared` in your environment to establish multiple secure, outbound-only, load-balanced links to Cloudflare. Simply put, Tunnel is what connects your network to Cloudflare.
+
+On the other side of this equation, your end users need to be able to easily connect to Cloudflare and, more importantly, your network. This connection is handled by Cloudflare WARP. This client can be rolled out to your entire organization in just a few minutes using your in-house MDM tooling, and it establishes a secure, WireGuard-based connection from your users’ devices to the Cloudflare network.
```
Comment (kennyj42) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md]: Do we mention wireGuard anywhere else in our docs? I think this is unnecessary.

**Review by kennyj42 — COMMENTED** [review]
I think we should remove the mention of WireGuard but could be mistaken. Don't think we bring that up anywhere else in docs

File: products/cloudflare-one/src/content/analytics/logs/index.md [hunk 2]
```diff
@@ -12,14 +12,13 @@ The Logs section of the [Teams dashboard](https://dash.teams.cloudflare.com/) al
...
+| --- | --- | --- | --- |
+| **Free** | 24 hours | 24 hours | 24 hours |
+| **Standard** | 30 days | 30 days | 30 days |
+| **Enterprise** | 180 days | 30 days | 30 days |
 
 ## How does Cloudflare use this data?
 
-For more information on how we use this data, please refer to our [Privacy Policy](https://www.cloudflare.com/en-gb/application/privacypolicy/).
-
+For more information on how we use this data, please refer to our [Privacy Policy](https://www.cloudflare.com/en-gb/application/privacypolicy/).
```
Comment (kodster28) [inline products/cloudflare-one/src/content/analytics/logs/index.md]:
```suggestion
For more information on how we use this data, please refer to our [Privacy Policy](https://www.cloudflare.com/application/privacypolicy/).
```

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/index.md [hunk 3]
```diff
@@ -1,28 +1,84 @@
...
+1. Create a tunnel with a user-friendly name to identify your network or environment. 
+
+    ```bash
+    $ cloudflared tunnel create acme-network
+    ```
+
+1. Finally, configure your tunnel with the IP/CIDR range of your private network. By doing this, you are creating a private network and making the WARP client aware that any requests to this IP range need to be routed to your new tunnel. 
+
+    ```bash
+    $ cloudflared tunnel route ip add 12.55.0.0/16
```
Comment (kodster28) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/index.md]:
```suggestion
    $ cloudflared tunnel route ip add 10.0.0.0/8
```

Comment (kodster28) [reply products/cloudflare-one/src/content/connections/connect-networks/private-net/index.md]: ^^ it's a reserved IP address for docs

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/index.md [hunk 4]
```diff
@@ -1,28 +1,84 @@
...
+    ```bash
+    $ cloudflared tunnel route ip show
+    ```
+
+1. Next, create a [configuration file](/connections/connect-apps/configuration/configuration-file) for the tunnel. The configuration file will be structured as follows: 
+
+    ```txt
+    tunnel: <Tunnel-UUID>
+    credentials-file: /root/.cloudflared/credentials-file.json
+    warp-routing:
```
Comment (kodster28) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/index.md:53]: Is the blank value for `warp-routing` intentional?

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/index.md [hunk 5]
```diff
@@ -1,28 +1,84 @@
...
+
+1. Run the tunnel. Traffic inside of your organization coming from enrolled WARP clients will be sent to this instance when the destination is your private IP range.
+
+    ```
+    $ cloudflared tunnel run acme-network
+    ```
+
+## (optional) Ensure that traffic can reach your network
+
+By default, Cloudflare for Teams [excludes traffic](/connections/connect-devices/warp/exclude-traffic/split-tunnels) to a specific set of destinations from WARP encryption. If you have not changed your default Split Tunnel settings, you can skip this step. If you have altered this configuration, for example by adding/removing entries to the Exclude list or by setting Split Tunnels mode to **Include**, make sure that traffic to the IP/CIDR you are associating with your private network is indeed set to be encrypted by WARP.
```
Comment (kodster28) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/index.md]:
```suggestion
By default, Cloudflare for Teams [excludes traffic](/connections/connect-devices/warp/exclude-traffic/split-tunnels) to a specific set of destinations from WARP encryption. If you have not changed your default Split Tunnel settings, you can skip this step. If you have altered this configuration, for example by adding or removing entries to the Exclude list or by setting Split Tunnels mode to **Include**, make sure that traffic to the IP/CIDR you are associating with your private network is set to be encrypted by WARP.
```

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
+---
+order: 4
+pcx-content-type: concept
+---
+
+# Private hostnames and IPs
+
+Building out a private network has two primary components: the infrastructure side, and the client side.
```
Comment (kodster28) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md]:
```suggestion
Building out a private network has two primary components: the infrastructure side and the client side.
```

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+The infrastructure side of the equation is powered by Cloudflare Tunnel, which connects your infrastructure to Cloudflare — whether that be a singular application, many applications, or an entire network segment. This is made possible by running `cloudflared` in your environment to establish multiple secure, outbound-only, load-balanced links to Cloudflare.
```
Comment (kodster28) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md]:
```suggestion
The infrastructure side is powered by Cloudflare Tunnel, which connects your infrastructure to Cloudflare — whether that be a singular application, many applications, or an entire network segment. This is made possible by running `cloudflared` in your environment to establish multiple secure, outbound-only, load-balanced links to Cloudflare.
```

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+On the client side, your end users need to be able to easily connect to Cloudflare and, more importantly, your network. This connection is handled by Cloudflare WARP. This client can be rolled out to your entire organization in just a few minutes using your in-house MDM tooling, and it establishes a secure connection from your users’ devices to the Cloudflare network.
```
Comment (kodster28) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md]:
```suggestion
On the client side, your end users need to be able to easily connect to Cloudflare and, more importantly, your network. This connection is handled by Cloudflare WARP. This client can be rolled out to your entire organization in just a few minutes using your in-house MDM tooling and it establishes a secure connection from your users’ devices to the Cloudflare network.
```

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+![Network diagram](../../../static/documentation/connections/private-ips-diagram.png)
+
+In this page, we will cover how to define your internal DNS resolver with Cloudflare for Teams, and how to create a new Private IP/Hostname application in Cloudflare Access.
```
Comment (kodster28) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md]:
```suggestion
In this page, we will cover how to define your internal DNS resolver with Cloudflare for Teams and how to create a new Private IP/Hostname application in Cloudflare Access.
```

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+1. Remain in Network Settings and scroll further down to **Local Domain Fallback**.
+1. Click **Manage**.
+1. Create a new Local Domain Fallback entry pointing to the internal DNS resolver. The rule in the following example instructs the WARP client to resolve all requests for `myorg.internal` through an internal resolver at `10.0.0.25` rather than attempting to resolve this publicly. 
+
+    ![Local domain fallback entry](../../../static/documentation/connections/local-domain-fallback-entry.png)
+
+<Aside type='note'>
+
+While on the Network Settings page, ensure that **Split Tunnels** are configured to include traffic to private IPs and hostnames in the traffic sent by WARP to Cloudflare. For guidance on how to do that, refer to [these instructions](/connections/connect-networks/private-net#optional-ensure-that-traffic-can-reach-your-network).
```
Comment (kodster28) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md]:
```suggestion
While on the Network Settings page, ensure that **Split Tunnels** are configured to include traffic to private IPs and hostnames in the traffic sent by WARP to Cloudflare. For guidance on how to do that, refer to [these instructions](/connections/connect-networks/private-net).
```

Comment (kodster28) [reply products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md]: That anchor no longer exists on the page

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+## Troubleshooting
+
+If you are encountering issues when setting up private IPs and hostnames for your network:
+
+* Ensure that `cloudflared` is connected to Cloudflare by visiting **Access** > **Tunnels** in the Cloudflare for Teams dashboard.
+
+* Ensure that end-user devices are enrolled into WARP by visiting `help.teams.cloudflare.com` on those devices. The page should return `Your network is fully protected`.
+
+* Ensure that the machine where `cloudflared` is running is allowed to egress via UDP to port 7844 to talk out to Cloudflare.
```
Comment (kodster28) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md]:
```suggestion
* Ensure that the machine where `cloudflared` is running is allowed to egress via UDP to port 7844 to communicate with Cloudflare.
```

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+* Double-check the precedence of your application policies in the **Gateway** > **Policies** > **Network policies** tab. Ensure that a more global Block or Allow policy will not supercede the application policies
```
Comment (kodster28) [inline products/cloudflare-one/src/content/connections/connect-networks/private-net/private-hostnames-ips.md]:
```suggestion
* Double-check the precedence of your application policies in the **Gateway** > **Policies** > **Network policies** tab. Ensure that a more global Block or Allow policy will not override the application policies.
```

**Review by kodster28 — APPROVED** [review]

File: products/cloudflare-one/src/content/connections/connect-networks/private-net/index.md (hunk 4 above)
Comment (abracchi-tw) [reply products/cloudflare-one/src/content/connections/connect-networks/private-net/index.md:53]: yep!

Comment (tjstansell) [reply products/cloudflare-one/src/content/connections/connect-networks/private-net/index.md:53]: I realize this is already merged, but the `enabled` line should be indented.  It's a property of the `warp-routing` above:
```
warp-routing:
    enabled: true
```
see https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/configuration-file#top-level-configurations

**Comment by tjstansell** [comment]
This PR also broke the link to [exposing a private network](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/private-networks) at https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/configuration-file#top-level-configurations

**Comment by abracchi-tw** [comment]
@tjstansell thanks for flagging, the link's being fixed here #3036

**Pull request merged**